### DISCUSSION TITLE
Transfer from Intel to Apple Silicon

### DISCUSSION THREAD

**Discussion by fishcharlie** [Homebrew/discussions#2370] [Getting started]
I recently transferred all my data from my Intel Mac to my Apple Silicon Mac. This transferred over all of my Homebrew data. When I launch Terminal I get:

```
env: bash: Bad CPU type in executable
env: bash: Bad CPU type in executable
env: bash: Bad CPU type in executable
```

I know that I can fix this by installing Rosetta 2, however that wouldn't really address the root problem. And someday if (or more likely when) Apple discontinues Rosetta 2, it will likely lead to this problem again.

When running: 

```
find `echo $PATH | tr ":" "\n"` -perm +1111 -exec file {} \; | grep i386`
```

There are a lot of results for python in `/usr/local/bin`.

I recently reinstalled Homebrew **hoping** that would wipe everything out and start things fresh for me. But a lot of remnants still exist it looks like.

What is the best way to uninstall and wipe out all of my old Intel Homebrew stuff? I'm completely fine with losing old Homebrew data at this point (it's easy to reinstall things). Just want to make sure I clean everything up.

**Comment by carlocab** [comment]
https://github.com/Homebrew/install#uninstall-homebrew

**Reply by fishcharlie** [reply]
@carlocab Will this remove the Intel version as well even tho I'm on a Apple Silicon Mac? And will it remove all packages installed by Homebrew?

**Reply by SMillerDev** [reply]
Yes, if you run it with Rosetta 
yes, I believe so. But you would not be able to use them without Rosetta anyways.

**Reply by carlocab** [reply]
Download the uninstall script and pass `--path=/usr/local` to uninstall your `/usr/local` installation of Homebrew entirely. See https://github.com/Homebrew/install/blob/cc7fe005e7449d29a064d82736ad5419a4f9da23/uninstall.sh#L184-L197.
